clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsClusterProvider.java

```diff
@@ -36,4 +46,46 @@ public EcsClusterProvider(Cache cacheView) {
   public Collection<EcsCluster> getAllEcsClusters() {
     return ecsClusterCacheClient.getAll();
   }
+
+  public Collection<Cluster> getAllEcsClustersDescription(String account, String region) {
+    List<String> clusterNames = new ArrayList<>();
+    List<Cluster> clusters = new ArrayList<>();
+    Collection<EcsCluster> ecsClusters = ecsClusterCacheClient.getAll();
+    AmazonECS client = getAmazonEcsClient(account, region);
+    if (ecsClusters.size() > 0 && ecsClusters != null) {
+      for (EcsCluster ecsCluster : ecsClusters) {
+        clusterNames.add(ecsCluster.getName());
+        if (clusterNames.size() % EcsClusterDescriptionMaxSize == 0) {
+          List<Cluster> describeClusterResponse = getDescribeClusters(client, clusterNames);
+          if (describeClusterResponse.size() > 0) {
+            clusters.addAll(describeClusterResponse);
+          }
+          clusterNames.clear();
+        }
+      }
+      if (clusterNames.size() % EcsClusterDescriptionMaxSize != 0) {
+        List<Cluster> describeClusterResponse = getDescribeClusters(client, clusterNames);
+        if (describeClusterResponse.size() > 0) {
+          clusters.addAll(describeClusterResponse);
+        }
+      }
+    }
+    return clusters;
+  }
+
+  private AmazonECS getAmazonEcsClient(String account, String region) {
+    NetflixECSCredentials credentials = credentialsRepository.getOne(account);
+    if (!(credentials instanceof NetflixECSCredentials)) {
+      throw new IllegalArgumentException("Invalid credentials:" + account + ":" + region);
+    }
+    return amazonClientProvider.getAmazonEcs(credentials, region, true);
+  }
+
+  private List<Cluster> getDescribeClusters(AmazonECS client, List<String> clusterNames) {
+    DescribeClustersRequest describeClustersRequest =
```

Comment:
What happens if you pass an empty list of clusters? Does this return an error?

Reply:
In that case we will get error in failures something like :

Reply:
In this case, it seems like the wrong approach to knowingly send an erroneous describeClusterRequest and expect a list of failures. It might make more sense to validate your inputs before you make the call in the first place (prevent unnecessary request/rate limiting, and it shouldn't be an error from the describe call if the input you provide this method is invalid)

Reply:
You can just return an emptyList if clusterNames is null or empty